snapcraft/__init__.py

```diff
@@ -377,6 +377,20 @@ def find_library(name):
     import ctypes.util
     ctypes.util.find_library = find_library
 
+
+def _get_version():
+    if _os.environ.get('SNAP_NAME') == 'snapcraft':
+        return _os.environ["SNAP_VERSION"]
+    try:
+        return pkg_resources.require('snapcraft')[0].version
+    except pkg_resources.DistributionNotFound:
+        return 'devel'
+
+
+# Set this early so that the circular imports aren't too painful
+__version__ = _get_version()
```

Comment:
You missed out a word, but I assume you're asking if I could move the definition of `__version__` above `_get_version`?

I cannot - this is a module level variable, it is evaluated at the time of importing the module, as such it cannot reference a function which hasn't yet been defined.

`_get_version` has to come before assignment to `__version__`